desc/geometry/core.py

```diff
@@ -17,18 +16,44 @@
 )
 from desc.grid import LinearGrid, QuadratureGrid, _Grid
 from desc.io import IOAble
-from desc.optimizable import Optimizable
+from desc.optimizable import Optimizable, optimizable_parameter
 
 
 class Curve(IOAble, Optimizable, ABC):
     """Abstract base class for 1D curves in 3D space."""
 
-    _io_attrs_ = ["_name", "shift", "rotmat"]
+    _io_attrs_ = ["_name", "_shift", "_rotmat"]
 
     def __init__(self, name=""):
-        self.shift = jnp.array([0, 0, 0]).astype(float)
-        self.rotmat = jnp.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]]).astype(float)
-        self.name = name
+        self._shift = np.array([0, 0, 0])
+        self._rotmat = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]]).flatten()
+        self._name = name
+
+    @optimizable_parameter
```

Comment:
When I implemented these the idea was they would not be optimized, just fixed at creation to make coilsets with shared geometry but fixed offset positions.

One main reason is that these parameters are redundant with the standard fourier coefficients so introduces some extra null space.

I'm open to making them optimizable but wanted to at least mention that

Reply:
Yeah I was going to bring this up at the dev meeting. They are redundant parameters, so making them optimizable means we have to be careful about the constraints. My first thought was to make these constants (like the transforms), but this option was less changes to the original code. 

I think the best solution would be to remove these completely. It might make it a little harder to initialize the geometry you want, but then everything else will be cleaner going forward.

Reply:
They actually were originally part of `transforms`: https://github.com/PlasmaControl/DESC/commit/65e4001663163621d085ab1381758e7413e0821c

I had to change it to allow vectorized evaluation of `CoilSet` stuff, where the transforms are the same for all members but params are different. We should talk tomorrow about the best way to handle these or possibly remove them.